packages/flutter_tools/lib/src/base/process.dart

```diff
@@ -278,21 +278,50 @@ abstract class ProcessUtils {
     );
   }
 
+  static Future<void> writelnToStdinUnsafe({
+    required IOSink stdin,
+    required String line,
+  }) async {
+    await _writeToStdinUnsafe(
+      stdin: stdin,
+      content: line,
+      isLine: true,
+    );
+  }
+
+  static Future<void> writeToStdinUnsafe({
```

Comment:
What about instead of having four of these public functions, we just have one, with a boolean parameter (that defaults to `true`) for whether to append newline, and then a nullable `onError` callback parameter--if `onError` is null then we unconditionally call `completer.completeError()` with caught errors?

My concern with four functions is that contributors will cargo cult the wrong one. There's a lot of cognitive complexity here understanding the differences. In a code review also, it's difficult to parse the differences.

Reply:
> In a code review also, it's difficult to parse the differences.

How so? Is this something I can address with doc comments? If the diff is difficult to parse, I'd recommend just reading the code directly.

Reply:
SGTM